File: ruby/app/fibonacci_generator.rb
class FibonacciGenerator
  def initialize(number)
    @fibonacci = Hash.new{ |h,k| h[k] = k < 2 ? k : h[k-1] + h[k-2] }
    @number = number
  end

  def calculate
    @fibonacci[@number]
  end
end

File: elixir/app/lib/fibonacci_generator.ex
defmodule FibonacciGenerator do
    defp comp_fib(0), do: [0 | 0]
    defp comp_fib(1), do: [1 | 0]

    defp comp_fib(n) do
        [h | t] = comp_fib(n-1)
        [h+t | h]
    end

    def calculate(n), do: hd(comp_fib(n))
end
